lib/lower/OpBuilder.cpp

```diff
@@ -64,3 +66,189 @@ void B::SymMap::pushShadowSymbol(Se::SymbolRef symbol, M::Value value) {
   assert(r.second && "unexpected insertion failure");
   (void)r;
 }
+
+M::Value B::OpBuilderHandler::getIntegerConstant(M::Type integerType,
+                                                 std::int64_t cst) {
+  return create<M::ConstantOp>(integerType,
+                               builder.getIntegerAttr(integerType, cst));
+}
+
+// LoopCreator implementation
+
+void B::LoopCreator::genLoop(M::Value lb, M::Value ub, M::Value step,
```

Comment:
There is already some code to generate a fir.loop, but having a builder for it is better. Just noting it, since we really need to refactor it and have The One True Way of doing it.

Reply:
OK, are you referring to other fir::Loop creation in Bridge.cpp here:
https://github.com/schweitzpgi/f18/blob/8a7694a5e692ea7a33ae6c5ed2e5f85e905436ac/lib/lower/Bridge.cpp#L371-L409
Or do you have another part of the code in mind ?

Reply:
Yes, the code around line 387 there. Ideally, I'd like to see a single path through burnside for building a `fir.loop` op. There may be reasons that's not possible, but abstraction is a good thing.

Reply:
I changed to use the `LoopBuilder` there too to keep things consistent.